src/plugins/data/public/ui/query_string_input/query_string_input.tsx

```diff
@@ -502,6 +528,34 @@ export class QueryStringInputUI extends Component<Props, State> {
     this.componentIsUnmounting = true;
   }
 
+  handleAutoHeight = () => {
+    if (this.inputRef !== null) {
+      this.inputRef.style.setProperty('height', `${this.inputRef.scrollHeight}px`, 'important');
+    }
```

Comment:
And then the autoheight is calced here. If for some reason a ref doesn't exist, it wouldn't fire, and the scroll would still exist. Can't think of a scenario where that would be the case.

Good luck, thanks for taking over.